### PR TITLE
Added Attachment Import/Export

### PR DESCRIPTION
**Pull request by stevenca**
- Added attachment import/export
- Cleaned up a few more built-ins that were stepped on
- Fixed some of the broken CLI commands

### CODE REVIEW

File: vmanage/api/device_templates.py [hunk 1]
```diff
@@ -258,13 +262,42 @@ def add_device_template(self, device_template):
                 payload['generalTemplates'] = self.generalTemplates_to_id(device_template['generalTemplates'])
             else:
                 raise Exception("No generalTemplates found in device template", data=device_template)
-            api = "template/device/feature"
-            url = self.base_url + api
+            url = f"{self.base_url}template/device/feature"
```
Comment (jlothian) [inline vmanage/api/device_templates.py:265]: do you ensure somewhere that base_url is terminated with a `/`?

File: vmanage/apps/files.py [hunk 2]
```diff
@@ -249,3 +250,82 @@ def import_policy_from_file(self, file, update=False, check_mode=False, push=Fal
...
+                        'template': device_config['template'],
+                        'variables': variable_dict
+                    }
+                    attachments_list.append(entry)
+
+        attachment_export = {'vmanage_attachments': attachments_list}
+        if export_file.endswith('.json'):
+            with open(export_file, 'w') as outfile:
+                json.dump(attachment_export, outfile, indent=4, sort_keys=False)
+        elif export_file.endswith('.yaml') or export_file.endswith('.yml'):
```
Comment (jlothian) [inline vmanage/apps/files.py:302]: can replace with :
elif export_file.endswith(('.yaml', 'yml)):

File: vmanage/apps/files.py [hunk 3]
```diff
@@ -249,3 +250,82 @@ def import_policy_from_file(self, file, update=False, check_mode=False, push=Fal
...
+
+    def import_attachments_from_file(self, file, update=False, check_mode=False, name_list=None, template_type=None):
+        vmanage_device_templates = DeviceTemplates(self.session, self.host, self.port)
+
+        template_data = {}
+        # Read in the datafile
+        if not os.path.exists(file):
+            raise Exception(f"Cannot find file {file}")
+        with open(file) as f:
+            if file.endswith('.yaml') or file.endswith('.yml'):
```
Comment (jlothian) [inline vmanage/apps/files.py:317]: ditto

File: vmanage/cli/export/attachments.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+@click.option('--file', '-f', 'output_file', help="Output file name", required=True)
+@click.option('--name', '-n', multiple=True)
+@click.option('--type',
+              '-t',
+              'device_type',
+              help="Device Type",
+              type=click.Choice(['controllers', 'vedges']),
+              default=None)
+@click.pass_obj
+def attachments(ctx, device_type, name, output_file):
```
Comment (jlothian) [inline vmanage/cli/export/attachments.py:15]: Try to keep the options in the function definition in the same order as they are declared with click.  It makes things less confusing.

File: vmanage/api/device_templates.py (hunk 1 above)
Comment (stevenca) [reply vmanage/api/device_templates.py:265]: No, but I am trying to move to this single line, then going to remove it and put move the slash to the url line from the base.  I'll create an issue for that.

File: tests/vmanage-attachments.yml [hunk 5]
```diff
@@ -0,0 +1,65 @@
+vmanage_attachments:
+-   host_name: vsmart1
+    device_type: vsmart
+    uuid: caecb73b-d163-402f-8aef-b3398394bcac
+    system_ip: 1.1.1.3
+    site_id: '1'
+    template: vsmart
+    variables:
+        vpn0_default_gateway: 192.133.184.1
```
[the remark this answers is not in the document]
Comment (stevenca) [reply tests/vmanage-attachments.yml:9]: I _could_ make fake one.  Let me think about that...

File: vmanage/cli/export/attachments.py (hunk 4 above)
Comment (stevenca) [reply vmanage/cli/export/attachments.py:15]: Fixed.

File: vmanage/apps/files.py (hunk 3 above)
Comment (stevenca) [reply vmanage/apps/files.py:317]: Fixed.

File: vmanage/apps/files.py (hunk 2 above)
Comment (stevenca) [reply vmanage/apps/files.py]: Fixed.

File: vmanage/api/device_templates.py (hunk 1 above)
Comment (stevenca) [reply vmanage/api/device_templates.py:265]: This cannot be done at the present time.

File: tests/cli.sh [hunk 6]
```diff
@@ -1,25 +1,17 @@
...
+path=`dirname $_`
+input="${path}/commands.txt"
+while IFS= read -r line
+do
+    printf "Running: $line..."
+    $line > /dev/null
+    if [ $? -eq 0 ]; then
+        echo OK
+    else
+        echo FAIL
```
[the remark this answers is not in the document]
Comment (stevenca) [reply tests/cli.sh:11]: We'll leave for a future PR

File: tests/vmanage-attachments.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (stevenca) [reply tests/vmanage-attachments.yml:9]: Cannot do this at the present time.

File: setup.py [hunk 7]
```diff
@@ -13,12 +13,12 @@
 
 setup(
     name="viptela",
-    version='0.2.0',
+    version='0.1.9',
```
[the remark this answers is not in the document]
Comment (stevenca) [reply setup.py]: Fixed.

File: tests/cli.sh [hunk 8]
```diff
@@ -1,25 +1,17 @@
 #!/bin/sh
```
[the remark this answers is not in the document]
Comment (stevenca) [reply tests/cli.sh]: Fixed.

**Pull request merged**